llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -3488,6 +3490,14 @@ void llvm::copyMetadataForLoad(LoadInst &Dest, const LoadInst &Source) {
       copyRangeMetadata(DL, Source, N, Dest);
       break;
     }
+    // Extended last-use / nontemporal hint on AMD GPUs
+    if (ID == Ctx.getMDKindID("amdpu.last.use"))
```

Comment:
Is this the one that we want to parameterize !nontemporal for? Probably should just do that?

Reply:
I think this is one of the owes I wanted to parametrize `nontemporal` for, yeah.

But `!lamdgpu.last.use` already landed so ... we might as well keep it plumbed through (though someone might want to do that for the gfx12 buffer op cache hints) until the nontemporal generality lands.